sdk/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/metrics/MetricPlugin.scala

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.speedy.metrics
+
+trait MetricPlugin {
+  type Result
```

Comment:
Using an inner type, rather than a type parameter, simplifies call site usage - i.e. call sites can use just the plugin type, rather than result and plugin type.